### DISCUSSION TITLE
[Nextjs 10] How does the image optimisation work?

### DISCUSSION THREAD

**Discussion by Francesco-Lanciana** [vercel/next.js#18343] [Help]
Hello!

I'm really keen to start using the Image component but I'm not sure if it fits my use case going off the existing documentation. I allow users to upload images and these are by default not publicly facing. I store these images on Google Cloud Storage. To access an image the client will first ask my backend for access, and if it determines the user is authorized it will generate and return a signed URL that I would assign to an img tag on the client. 

Is this something that the Image component can optimise? I'm also really unclear on how this optimisation is occurring. Is it happening at runtime? Do you have to pay a one time penalty the first time you request the image or does it not optimise the image on that request? Where are these optimised images stored? How is it optimising images that aren't hosted on Vercel?

Thanks in advance for the help! And sorry for the barrage of questions

**Comment by timneutkens** [comment]
Pending a stable release, but we've added some info here: https://nextjs.org/docs/tag/canary/basic-features/image-optimization#caching

**Comment by Francesco-Lanciana** [comment]
Thanks Tim,

That definitely helps for when I'm hosting the image myself but it doesn't really address how to use this for user uploaded images that are stored in a bucket like GCS and aren't public facing. 

Is there a way to dynamically optimise these types of images or do we need an external provider for this use case?

Also I was under the impression that image optimisation was stable since it was included in Next 10.0?

**Reply by timneutkens** [reply]
> Also I was under the impression that image optimisation was stable since it was included in Next 10.0?

I was referring to that docs update that included the section on caching. The component/optimization is stable.

> Is there a way to dynamically optimise these types of images or do we need an external provider for this use case?

If it's behind auth or some kind of mechanism that does not allow public serving you'll have to implement an API route to serve the files and then use that url in `next/image` `src`

**Comment by binajmen** [comment]
Hi @Francesco-Lanciana,

I would be interested to hear your experience when combining GCS images with Next image optimization.

I'm facing the same question that is: can I use next image optimization with GCS images. In my case, the images are public, so I already have the signed url available for the end-user.

However, I have sporadically 503 error. Some pictures are loaded, other not (503 response). If I refresh, it succeeds (not always). Overall, not a stable experience.

![Screenshot 2021-04-13 at 15 01 16](https://user-images.githubusercontent.com/15611419/114556584-27503880-9c69-11eb-8d08-c9eeba8327e5.png)
 
@timneutkens, is there a timeout or a reason why it would sometimes fail? If I open the url in the query parameters used in `_next/image?url=..`, the image is accessible. I'm running NextJS on Cloud Run, and have no issues with static images nor other assets used to run the app.

**Reply by ashmore11** [reply]
@binajmen Did you find a solution to this issue? Running into it myself currently.

**Reply by binajmen** [reply]
Unfortunately, no :(

**Reply by ashmore11** [reply]
@timneutkens has your team encountered this issue with Google Cloud Storage? Currently image optimisation is unusable for me because of this. Happy to open a report if it's not already on your radar. Cheers!

**Reply by ashmore11** [reply]
@timneutkens I'm still running into issues with this and can't seem to find any workarounds. I'd rather not have to drop GCS completely as it's already pretty deeply integrated within my application. From what I know a Google Engineer is working on the image component so surely they've encountered this issue? Would love an update / glimmer of hope 😅 Thanks!

@binajmen Any luck on your side?

**Reply by ashmore11** [reply]
@binajmen @timneutkens Any luck with this one? I'm still seeing issues with the latest version of Next unfortunately : (

**Reply by dazulu** [reply]
Hit this issue today with next/image and sources coming from Google Cloud Storage. "Got around it" by adding `loader={({src})=>src}` but may as well just use an img tag at that point :)

**Reply by ocelot2123** [reply]
@ashmore11 do you have an update on this? I seem to be facing the same issues now. Where can I find more info about the image component that is being worked on? Any help would be greatly appreciated, thanks!

**Reply by tombohub** [reply]
Is this happening only with GCS?
